src/main/java/duke/Command.java

```diff
@@ -0,0 +1,40 @@
+package duke;
+
+/**
+ * Represents a Command
+ */
+public class Command {
+
+    /**
+     * Type of the Command
+     */
+    public Duke.CommandType type;
+
+    /**
+     * Stores the arguments required for running of command
+     */
+    public String[] args;
+
+    /**
+     * Constructor for Commands which do not require arguments
+     *
+     * @param newType Type of the Command
+     */
+    public Command(Duke.CommandType newType) {
```

Comment:
Maybe you can rename 'newType' as 'type' because this.type already differentiates between class fields and method parameters?